code/processes/tickerlogreplay.q

```diff
@@ -345,20 +323,52 @@ merge:{[dir;pt;tablename;mergelimits;h]
  if[gc;.gc.run[]];
  };
 
+// Get all logs that need to be replayed
+getlogs:{
+  if[all null (.replay.tplogfile;.replay.tplogdir);.err.ex[`getlogs;"No log file or directory given."];1];
+  
+  // If in segmented mode, grab meta table and grab log file names from there, return early
+  if[.replay.segmentedmode;
+    metatable:get .Q.dd[hsym first .replay.tplogfile;`stpmeta];
+    :exec distinct logname from metatable where any each tbls in .replay.tablestoreplay;
+   ];
+
+  // Use tplogfile/dir to return log names
+  $[not null .replay.tplogfile;
+    $[()~key hsym f:.replay.tplogfile;
+      .err.ex[`replayinit;"specified tplogfile ",string[f]," does not exist";2];
+      enlist hsym .replay.tplogfile
+     ];
+    $[()~key hsym d:.replay.tplogdir;
+      .err.ex[`replayinit;"specified tplogfiles ",string[d]," do not exist";3];
+      raze ` sv' tplogdir,/:key tplogdir:hsym .replay.tplogdir
+     ]
+   ]
+ };
 
-\d .
-//load the sort csv
-.sort.getsortcsv[.replay.sortcsv]
-
-
-if[.replay.autoreplay;
-  // grab stp meta table before replaying logs if in segmentedmode
-  stpmeta:([]seq:(), logname:(), start:(), end:(), tbls:(), msgcount:(), schema:(), additional:());
-  .lg.o[`replay;"replay starting from script by default"];
-  // expand stp log directories if using segmentedmode
-  if[.replay.segmentedmode;.replay.logstoreplay:.replay.expandstplogs[.replay.logstoreplay]];
+// Set up log replay list and clean HDB if necessary, kick off replay
+initandrun:{
+  .lg.o[`initandrun;"Initialising and beginning replay"];
+  .replay.tablestoreplay:$[`all~first .replay.tablelist;tables[];.replay.tablelist,()];
+  .replay.logstoreplay:$[count f:.replay.getlogs[];f;.err.ex[`initandrun;"No log files found";4]];
+  .lg.o[`initandrun;"Replaying the following log(s): ",csv sv 1_'string .replay.logstoreplay];
+
+  // If all log files have the same date, set as a variable, error out if not
+  $[all 0=1_deltas l:"D"$$[.replay.segmentedmode;-6_-14#;-10#] each string .replay.logstoreplay;
+    .replay.replaydate:first l;
+    [.lg.o[`replay;m:"Cannot replay logs from different dates!"];'m]
```

Comment:
I think this is taking away a bit of functionality that we had previously to replay a directory of multiple log files (when running in the old mode)